### PR TITLE
add benchmark for ChangeNotifier.notifyListeners

### PR DESCRIPTION
**Pull request by rrousselGit**
## Description

Related to https://github.com/flutter/flutter/pull/62330
This adds a benchmark for ChangeNotifier.notifyListeners, so that we can track performance improvements.

## Related Issues

NaN

## Tests

NaN

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by goderbauer** [comment]
The benchmark also has to be added here so it runs in our test harness on every commit: https://github.com/flutter/flutter/blob/6e0bc30e84de96294f0141c3813872b3e9156ed6/dev/devicelab/lib/tasks/microbenchmarks.dart#L56

We should also benchmark adding and removing listeners to ensure that any change doesn't make those operations slow.

**Comment by rrousselGit** [comment]
done

**Comment by goderbauer** [comment]
Can you please fix the analyzer warning about the white-space?

Most of the test failures look unrelated. Presumably, those can be fixed by rebasing this onto the latest master.

File: dev/benchmarks/microbenchmarks/lib/foundation/change_notifier_bench.dart [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      addListenerWatch.stop();
+      notifyListenersWatch.start();
+
+      notifier.notify();
+
+      notifyListenersWatch.stop();
+      removeListenerWatch.start();
+
+      // Remove listeners in reverse order to evaluate the worse-case scenario:
+      // the listener remove is the last listener
```
Comment (goderbauer) [inline dev/benchmarks/microbenchmarks/lib/foundation/change_notifier_bench.dart:68]: remove -> removed

**Review by goderbauer — APPROVED** [review]
LGTM once all the checks pass

**Pull request merged**